src/value/mod.rs

```diff
@@ -838,6 +838,395 @@ mod generic_value_tests {
         }
     }
 
+    struct CaseStruct<E: Element> {
+        eq_annotations: Vec<E>,
+        ne_annotations: Vec<E>,
+    }
+
+    fn struct_with_local_sid_case<E: Element>() -> CaseStruct<E>
+    where
+        E::Builder: Builder<SymbolToken = E::SymbolToken>,
+    {
+        CaseStruct {
+            eq_annotations: vec![
+                E::Builder::new_struct(
+                    vec![(
+                        E::SymbolToken::local_sid_token(21),
+                        E::Builder::new_string("hello"),
+                    )]
+                    .into_iter(),
+                ),
+                // SymbolToken with local SID and no text are equivalent to each other and to SID $0
+                E::Builder::new_struct(
+                    vec![(
+                        E::SymbolToken::local_sid_token(22),
```

Comment:
Highlighting this as part of our offline discussion: this is a case where we're constructing an instance of a type that implements `SymbolToken` and our expectation is that "we didn't provide text, therefore there is no text."